python/pfs/drp/stella/lsf.py

```diff
@@ -0,0 +1,688 @@
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        stdev : `float`
+            Standard deviation of the LSF at the nominated position.
+        """
+        return self.computeKernel(center).computeStdev()
+
+    def computeApertureFlux(self, radius, point=None):
+        """Return the flux within the nominated radius
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        flux : `float`
+            Flux within the nominated radius.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        kernel = self.computeKernel(point.getX())
+        intRadius = int(radius)
+
+        flux = trapz([kernel[ii] for ii in range(-intRadius, intRadius + 1)])
+        if radius != intRadius:
+            residual = radius - intRadius
+            left, right = kernel.interpolate([-radius, radius])
+            flux += 0.5*residual*(kernel[-intRadius - 1] + kernel[-intRadius])
+            flux += 0.5*residual*(kernel[intRadius + 1] + kernel[intRadius])
+        return flux
+
+    def getLocalKernel(self, point=None):
+        """Return a 2D ``FixedKernel`` representing the Lsf at a point
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        kernel : `lsst.afw.math.FixedKernel`
+            Kernel representing the Lsf at the nominated point.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        kernel = self.computeKernel(point.getX())
+        image = ImageD(kernel.values[np.newaxis, :], True, Point2I(-kernel.center, 0))
+        return FixedKernel(image)
+
+    def getAveragePosition(self):
+        """Return the average position for which the Lsf is defined
+
+        Returns
+        -------
+        position : `lsst.geom.Point2D`, optional
+            Average position for which the Lsf is defined.
+        """
+        return Point2D(0.5*self.length, 0)
+
+    def computeBBox(self, point=None):
+        """Return the bounding box of the kernel image
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        bbox : `lsst.afw.geom.Box2I`
+            Bounding box for the kernel image.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        return self.computeKernelImage(point.getX()).getBBox()
+
+    @classmethod
+    def readFits(cls, filename):
+        """Read from file
+
+        Though the method name says "FITS", we read a pickle file, because
+        pickling takes care of subclasses for us.
```

Comment:
Hmm.  We cannot write anything that we wish to keep as pickle files as they become unreadable if we change things in the implementation.  Probably OK in the short run, but it must be flagged.

Reply:
DAMD-78. I don't think it makes sense to schedule this until we've had a chance to play around with the `Lsf` classes here and get a feel for what the datamodel should be.